src/diffusers/pipelines/stable_diffusion/convert_from_ckpt.py

```diff
@@ -828,13 +856,50 @@ def load_pipeline_from_original_stable_diffusion_ckpt(
             Whether the attention computation should always be upcasted. This is necessary when running stable
             diffusion 2.1.
         device (`str`, *optional*, defaults to `None`):
-            The device to use. Pass `None` to determine automatically. :param from_safetensors: If `checkpoint_path` is
-            in `safetensors` format, load checkpoint with safetensors instead of PyTorch. :return: A
-            StableDiffusionPipeline object representing the passed-in `.ckpt`/`.safetensors` file.
+            The device to use. Pass `None` to determine automatically.
+        from_safetensors (`str`, *optional*, defaults to `None`):
+            If `checkpoint_path` is in `safetensors` format, load checkpoint with safetensors instead of PyTorch.
+            Pass `None` to determine automatically based on checkpoint file extension
+        pipeline_class (`str`, *optional*, defaults to `None`):
+            The pipeline class to use. Pass `None` to determine automatically.
+        requires_safety_checker (`str`, *optional*, defaults to `True`):
+            Whether to use the safety checker for original stable diffusion checkpoints.
     """
+    from diffusers import StableDiffusionPipeline, LDMTextToImagePipeline, PaintByExamplePipeline
+
     if prediction_type == "v-prediction":
         prediction_type = "v_prediction"
 
+    if from_safetensors is None:
+        from_safetensors = checkpoint_path.rsplit(".", 1)[-1] == "safetensors"
+
+    checkpoint_path = Path(checkpoint_path)
+
+    # Code based on diffusers.pipelines.pipeline_utils.DiffusionPipeline.from_pretrained
+    if not os.path.isfile(checkpoint_path):
+        # get repo_id and (potentially nested) file path of ckpt in repo
+        repo_id = str(Path().joinpath(*checkpoint_path.parts[:2]))
+        file_path = str(Path().joinpath(*checkpoint_path.parts[2:]))
+
+        cache_dir = kwargs.pop("cache_dir", DIFFUSERS_CACHE)
+        resume_download = kwargs.pop("resume_download", False)
+        proxies = kwargs.pop("proxies", None)
+        local_files_only = kwargs.pop("local_files_only", HF_HUB_OFFLINE)
+        use_auth_token = kwargs.pop("use_auth_token", None)
+        revision = kwargs.pop("revision", None)
+
+        cached_folder = snapshot_download(
```

Comment:
actually we should use `snapshot_download` here since we're just downloading a single .ckpt file. Could you instead make use of: https://huggingface.co/docs/huggingface_hub/v0.12.1/en/package_reference/file_download#huggingface_hub.hf_hub_download

More specifically the following should work:

```py
file = hf_hub_download(
            repo_id,
            filename=file_path,
            cache_dir=cache_dir,
            resume_download=resume_download,
            proxies=proxies,
            local_files_only=local_files_only,
            use_auth_token=use_auth_token,
            revision=revision,
        )
```

Reply:
Yes will do, thanks for the link!